streams/jvm/src/main/scala/scalaz/zio/stream/platform.scala

```diff
@@ -0,0 +1,51 @@
+package scalaz.zio.stream
+
+import java.io.{ InputStream, OutputStream }
+
+import scalaz.zio._
+import scalaz.zio.blocking._
+
+trait ZStreamPlatformSpecific {
+
+  /**
+   * Uses the provided `TaskR` value to create a [[ZStream]] of byte chunks, backed by
+   * the resulting `InputStream`. When data from the `InputStream` is exhausted,
+   * the stream will close it.
+   */
+  def fromInputStream[R](
+    is: TaskR[R, InputStream],
```

Comment:
An alternative (I am not sure if it is a good idea) is this could take a Managed InputStream and then we could provide a helper to create that, e.g. `ZManaged.fromInputStream(is: ZIO[R, E, InputStream]): ZManaged[R, E, InputStream): ZManaged[R, E, InputStream]` (possibly more convenient, `effectInputStream(is: => InputStream): ZManaged[Any, Throwable, InputStream]`.

Reply:
I need to use it a bit to form an opinion :-) So I'll keep the current signature and follow up later on with improvements.